capplet/msm-desktop-app-dialog.c

```diff
@@ -0,0 +1,314 @@
+    selected = gtk_list_box_get_selected_row (GTK_LIST_BOX (dialog->listbox));
+
+    if (selected != NULL && gtk_widget_get_mapped (GTK_WIDGET (selected)))
+    {
+        gtk_dialog_set_response_sensitive (GTK_DIALOG (dialog), GTK_RESPONSE_OK, TRUE);
+    }
+    else
+    {
+        gtk_dialog_set_response_sensitive (GTK_DIALOG (dialog), GTK_RESPONSE_OK, FALSE);
+    }
+}
+
+static gint
+list_sort_app_func (GtkListBoxRow *row1, GtkListBoxRow *row2, gpointer user_data)
+{
+    GAppInfo *app1, *app2;
+
+    app1 = g_object_get_data (G_OBJECT (row1), "appinfo");
+    app2 = g_object_get_data (G_OBJECT (row2), "appinfo");
+
+    return g_utf8_collate (g_app_info_get_name (app1), g_app_info_get_name (app2));
+}
+
+static void
+list_header_app_func (GtkListBoxRow *row, GtkListBoxRow *before, gpointer user_data)
+{
+
+    if (before != NULL && (gtk_list_box_row_get_header (row) == NULL))
+    {
+        gtk_list_box_row_set_header (row,
+                                     gtk_separator_new (GTK_ORIENTATION_HORIZONTAL));
+    }
+}
+
+static gboolean
+list_filter_app_func (GtkListBoxRow *row,
+                      gpointer       user_data)
+{
+    MsmDesktopAppDialog *dialog = MSM_DESKTOP_APP_DIALOG (user_data);
+    GAppInfo *app;
+    GString  *s1, *s2;
+
+    app = g_object_get_data (G_OBJECT (row), "appinfo");
+
+    s1 = g_string_new (gtk_entry_get_text (GTK_ENTRY (dialog->entry)));
+    s2 = g_string_new (g_app_info_get_name (app));
+
+    g_string_ascii_up (s1);
+    g_string_ascii_up (s2);
+
+    if (strstr (s2->str, s1->str) != NULL)
+        return TRUE;
+
+    return FALSE;
+}
+
+static void
+on_row_selected (GtkListBox          *box,
+                 GtkListBoxRow       *row,
+                 MsmDesktopAppDialog *dialog)
+{
+    if (row != NULL && gtk_widget_get_mapped (GTK_WIDGET (row)))
+    {
+        gtk_dialog_set_response_sensitive (GTK_DIALOG (dialog), GTK_RESPONSE_OK, TRUE);
+    }
+    else
+    {
+        gtk_dialog_set_response_sensitive (GTK_DIALOG (dialog), GTK_RESPONSE_OK, FALSE);
+    }
+}
+
+static void
+setup_dialog (MsmDesktopAppDialog *dialog)
+{
+    GtkStyleContext *context;
+    GtkWidget   *content_area;
+    GtkWidget   *searchbar;
+    GtkWidget   *searchbutton;
+    GtkWidget   *searchimage;
+    GtkWidget   *headerbar;
+    GtkWidget   *sw;
+
+    content_area = gtk_dialog_get_content_area (GTK_DIALOG (dialog));
+    gtk_container_set_border_width (GTK_CONTAINER (content_area), 6);
+    gtk_window_set_position (GTK_WINDOW (dialog), GTK_WIN_POS_CENTER);
+    gtk_widget_set_size_request (GTK_WIDGET (dialog), 600, 400);
+    gtk_window_set_modal (GTK_WINDOW (dialog), TRUE);
+
+    dialog->entry = gtk_search_entry_new ();
+    g_signal_connect (dialog->entry,
+                     "search-changed",
+                      G_CALLBACK (on_search_entry_changed),
+                      dialog);
+
+    gtk_entry_set_placeholder_text (GTK_ENTRY (dialog->entry), _("Search Applications…"));
```

Comment:
The text isn't visible in non-csd version, and in headerbar vsersion you need to click on the button to make the text visible.

Reply:
In non-csd dialog the text is visible when you drag the window o_0

Reply:
Got it, when the entry got focus than the text goes away.